From 673e9988fc912bb73ce26a951e1b27a6249cd0d5 Mon Sep 17 00:00:00 2001
From: chandu078 <chandudyavanapelli03@gmail.com>
Date: Mon, 23 Feb 2026 18:35:51 +0000
Subject: [PATCH 1/3] compat: Provide libbluetooth_audio_session_aidl shim

BluetoothAudioSession::IsSessionReady() was changed in
https://android.googlesource.com/platform/hardware/interfaces/+/31c91b54f0e7308c13c80b098941e0811f7e90d2%5E%21/#F1.

This shim restores the old symbol for backwards compatibility.

Change-Id: Ib65cf1a5dd770502d872e7f82c819dc5c341d9c4
---
 Android.bp                                         | 11 +++++++++++
 .../BluetoothAudioSession.cpp                      | 14 ++++++++++++++
 .../BluetoothAudioSession.h                        | 14 ++++++++++++++
 3 files changed, 39 insertions(+)
 create mode 100644 libbluetooth_audio_session_aidl/BluetoothAudioSession.cpp
 create mode 100644 libbluetooth_audio_session_aidl/BluetoothAudioSession.h

diff --git a/Android.bp b/Android.bp
index 1283248..94adc59 100644
--- a/Android.bp
+++ b/Android.bp
@@ -72,6 +72,17 @@ cc_library_shared {
     vendor: true,
 }
 
+cc_library_shared {
+    name: "libbluetooth_audio_session_aidl_shim",
+    shared_libs: [
+        "libbluetooth_audio_session_aidl",
+    ],
+    srcs: [
+        "libbluetooth_audio_session_aidl/BluetoothAudioSession.cpp",
+    ],
+    vendor: true,
+}
+
 cc_defaults {
     name: "libgui_shim_defaults",
     shared_libs: ["libutils"],
diff --git a/libbluetooth_audio_session_aidl/BluetoothAudioSession.cpp b/libbluetooth_audio_session_aidl/BluetoothAudioSession.cpp
new file mode 100644
index 0000000..f550390
--- /dev/null
+++ b/libbluetooth_audio_session_aidl/BluetoothAudioSession.cpp
@@ -0,0 +1,14 @@
+/*
+ * SPDX-FileCopyrightText: The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "BluetoothAudioSession.h"
+
+namespace aidl::android::hardware::bluetooth::audio {
+
+bool BluetoothAudioSession::IsSessionReady() {
+    return IsSessionReady(false);
+}
+
+}  // namespace aidl::android::hardware::bluetooth::audio
diff --git a/libbluetooth_audio_session_aidl/BluetoothAudioSession.h b/libbluetooth_audio_session_aidl/BluetoothAudioSession.h
new file mode 100644
index 0000000..bfae10a
--- /dev/null
+++ b/libbluetooth_audio_session_aidl/BluetoothAudioSession.h
@@ -0,0 +1,14 @@
+/*
+ * SPDX-FileCopyrightText: The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+namespace aidl::android::hardware::bluetooth::audio {
+
+class BluetoothAudioSession {
+public:
+    bool IsSessionReady();
+    bool IsSessionReady(bool is_primary_hal);
+};
+
+}  // namespace aidl::android::hardware::bluetooth::audio
